### PR TITLE
fix(@desktop/wallet): Implement missing unpreferred chains

### LINKED ISSUE #8530: Implement missing unprefered chains
### Description

if not better path is available, then the use should be shown path with unprefered chains as show in design https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=3231%3A181254&t=dH45Y1y30MclwOh1-4

Also if no prefix is provided we should only show user option to send over mainnet

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes #8530

### What does the PR do

Implement unpreferred and preferred networks 

### Affected areas

Wallet
### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

image

### CODE REVIEW

File: ui/imports/shared/views/NetworkCardsComponent.qml [hunk 1]
```diff
@@ -49,16 +50,19 @@ Item {
...
             }
         }
+
+        function draw() {
+            canvas.clear()
+            canvas.requestPaint()
+        }
     }
 
+    onVisibleChanged: d.draw()
```
Comment (caybro) [inline ui/imports/shared/views/NetworkCardsComponent.qml:60]:
```suggestion
    onVisibleChanged: if (visible) d.draw()
```

File: ui/imports/shared/views/NetworkCardsComponent.qml [hunk 2]
```diff
@@ -131,12 +135,12 @@ Item {
...
                     property bool isOnBestRoute: false
                     property double amountToReceive: 0
+                    property bool preferred: store.preferredChainIds.includes(model.chainId)
                     primaryText: model.chainName
                     secondaryText: LocaleUtils.numberToLocaleString(amountToReceive)
-                    tertiaryText: ""
-                    // To-do preferred in not something that is supported yet
-                    state: root.errorMode ? "error" : "default"
-                    // opacity: preferred ? 1 : 0
+                    tertiaryText: state === "unpreferred"  ? qsTr("unpreferred").toUpperCase() : ""
```
Comment (caybro) [inline ui/imports/shared/views/NetworkCardsComponent.qml:141]: `qsTr("unpreferred").toUpperCase()` 

No just no, some languages don't even have uppercase letters :D

File: ui/imports/shared/views/NetworksAdvancedCustomRoutingView.qml [hunk 3]
```diff
@@ -50,12 +50,17 @@ ColumnLayout {
                     wrapMode: Text.WordWrap
                 }
                 StatusButton {
+                    id: preferredToggleButton
+                    property bool btnChecked: false
```
Comment (caybro) [inline ui/imports/shared/views/NetworksAdvancedCustomRoutingView.qml:54]: Why an extra property? You already have a builtin `checked` property available

File: ui/imports/shared/views/NetworksAdvancedCustomRoutingView.qml [hunk 4]
```diff
@@ -50,12 +50,17 @@ ColumnLayout {
...
-                    verticalPadding: 0
+                    verticalPadding: -1
                     size: StatusBaseButton.Size.Small
-                    icon.name: "hide"
-                    text: qsTr("Show Unpreferred Networks")
+                    icon.name: btnChecked ? "show" : "hide"
+                    icon.height: 16
+                    icon.width: 16
+                    text: btnChecked ? qsTr("Hide Unpreferred Networks"): qsTr("Show Unpreferred Networks")
+                    onClicked: btnChecked = !btnChecked
```
Comment (caybro) [inline ui/imports/shared/views/NetworksAdvancedCustomRoutingView.qml:63]: `onToggled` then

File: ui/imports/shared/views/NetworkCardsComponent.qml (hunk 2 above)
Comment (caybro) [reply ui/imports/shared/views/NetworkCardsComponent.qml:141]:
```suggestion
                    tertiaryText: state === "unpreferred" ? qsTr("UNPREFERRED") : ""
```

**Review by caybro — CHANGES_REQUESTED** [review]
Minor UI things :police_officer:

File: ui/imports/shared/views/NetworkCardsComponent.qml (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/views/NetworkCardsComponent.qml:60]: Aaah thanks! I missed it :)

File: ui/imports/shared/views/NetworkCardsComponent.qml (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/views/NetworkCardsComponent.qml:141]: hahahaha yes! again thanks for finding this!

**Review by saledjenic — APPROVED** [review]

File: ui/imports/shared/views/NetworkCardsComponent.qml (hunk 1 above)
Comment (caybro) [reply ui/imports/shared/views/NetworkCardsComponent.qml:60]: Wrong line!

File: ui/imports/shared/views/NetworkCardsComponent.qml [hunk 5]
```diff
@@ -49,16 +50,19 @@ Item {
...
         }
+
+        function draw() {
+            canvas.clear()
+            canvas.requestPaint()
+        }
     }
 
+    onVisibleChanged: d.draw()
+    onBestRoutesChanged: if(visible) d.draw()
```
Comment (caybro) [inline ui/imports/shared/views/NetworkCardsComponent.qml:61]: Move the `if (visible)` check above

File: ui/imports/shared/views/NetworksAdvancedCustomRoutingView.qml [hunk 6]
```diff
@@ -50,12 +50,17 @@ ColumnLayout {
...
+                    verticalPadding: -1
+                    checkable: true
                     size: StatusBaseButton.Size.Small
-                    icon.name: "hide"
-                    text: qsTr("Show Unpreferred Networks")
+                    icon.name: checked ? "show" : "hide"
+                    icon.height: 16
+                    icon.width: 16
+                    text: checked ? qsTr("Hide Unpreferred Networks"): qsTr("Show Unpreferred Networks")
+                    onCheckedChanged: if(!checked) store.addUnpreferredChainsToDisabledChains()
```
Comment (caybro) [inline ui/imports/shared/views/NetworksAdvancedCustomRoutingView.qml:63]: `onToggled`!

**Review by caybro — CHANGES_REQUESTED** [review]

File: ui/imports/shared/views/NetworksAdvancedCustomRoutingView.qml (hunk 6 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/views/NetworksAdvancedCustomRoutingView.qml]: @caybro , that was wasteful of me. done now :)

**Review by caybro — APPROVED** [review]
Thanks :)

**Pull request merged**